spektrum/runner.py

```diff
@@ -92,21 +92,48 @@ def run(self, search_paths, module_name=None, metadata=None, test_names=None, ex
 
         return self.reporting.success
 
+    def _map_classes(self, classes, class_dict, parent):
+        for cls in classes:
+            cur_node = f'{parent}.{(cls.__name__)}'
+            class_dict[cur_node] = cls
+            children = find_children(cls)
+            self._map_classes(children, class_dict, cur_node)
+
     def filter_by_module_name(self, classes, name):
         found = [
             cls
             for cls in classes
             if name in '{}.{}'.format(cls.__module__, cls.__name__)
         ]
 
-        # Only search children if the class cannot be found at the package lvl
         if not found:
-            children = [find_children(cls) for cls in classes]
-            found = [
-                cls
-                for cls in itertools.chain.from_iterable(children)
-                if name in '{}.{}'.format(cls.__module__, cls.__name__)
-            ]
+            try:
+                module = next(
+                    cls
+                    for cls in classes
+                    if cls.__module__ in name
+                )
+                class_dict = {}
```

Comment:
Nitpicking, but if we're just passing in an empty dict as a parameter to populate, I think this method should probably be written to return a dict instead of passing by reference. When called recursively, you can just use the dict `.update()` method with the dict of children from the recursive call.